OK, AppVeyor failed on added test:

```
======================================================================
FAIL: test_issue20884 (test.test_cmd_line_script.CmdLineTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\cpython\lib\test\test_cmd_line_script.py", line 425, in test_issue20884
    rc, out, err = assert_python_ok(script_name)
  File "C:\projects\cpython\lib\test\support\script_helper.py", line 157, in assert_python_ok
    return _assert_python(True, *args, **env_vars)
  File "C:\projects\cpython\lib\test\support\script_helper.py", line 143, in _assert_python
    res.fail(cmd_line)
  File "C:\projects\cpython\lib\test\support\script_helper.py", line 70, in fail
    raise AssertionError("Process return code is %d\n"
AssertionError: Process return code is 1
command line: ['C:\\projects\\cpython\\PCbuild\\amd64\\python.exe', '-X', 'faulthandler', '-I', 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmpnejtdxss\\issue20884.py']
stdout:
---
---
stderr:
---
File "C:\Users\appveyor\AppData\Local\Temp\1\tmpnejtdxss\issue20884.py", line 1
SyntaxError: encoding problem: iso-8859-1
---
----------------------------------------------------------------------
```

--- a/Doc/c-api/veryhigh.rst
+++ b/Doc/c-api/veryhigh.rst
@@ -110,7 +110,7 @@ the same library that the Python runtime is using.
    closed before PyRun_SimpleFileExFlags returns.
 
    .. note::
-      On Windows, `FILE *fp` should be opened as binary mode (e.g. `fopen(filename, "rb")`.
+      On Windows, *fp* should be opened as binary mode (e.g. ``fopen(filename, "rb")``.
       Otherwise, Python may not handle script file with LF line ending correctly.
 
 

--- a/Doc/c-api/veryhigh.rst
+++ b/Doc/c-api/veryhigh.rst
@@ -109,6 +109,10 @@ the same library that the Python runtime is using.
    (:func:`sys.getfilesystemencoding`).  If *closeit* is true, the file is
    closed before PyRun_SimpleFileExFlags returns.
 
+   .. note::
+      On Windows, `FILE *fp` should be opened as binary mode (e.g. `fopen(filename, "rb")`.
+      Otherwise, Python may not handle script file with LF line ending correctly.
+
 
 .. c:function:: int PyRun_InteractiveOne(FILE *fp, const char *filename)
 

--- a/Misc/NEWS.d/next/Core and Builtins/2019-03-29-18-47-50.bpo-20844.ge-7SM.rst
+++ b/Misc/NEWS.d/next/Core and Builtins/2019-03-29-18-47-50.bpo-20844.ge-7SM.rst
@@ -1,0 +1,2 @@
+Fix running script with encoding cookie and LF line ending
+may fail on Windows.

--- a/Modules/main.c
+++ b/Modules/main.c
@@ -283,7 +283,7 @@ static int
 pymain_run_file(_PyCoreConfig *config, PyCompilerFlags *cf)
 {
     const wchar_t *filename = config->run_filename;
-    FILE *fp = _Py_wfopen(filename, L"r");
+    FILE *fp = _Py_wfopen(filename, L"rb");
     if (fp == NULL) {
         char *cfilename_buffer;
         const char *cfilename;

